infrastructure/pyd/make_object.d

```diff
@@ -445,6 +450,21 @@ T python_to_d(T) (PyObject* o) {
     } else static if (is(T == void)) {
         if (o != cast(PyObject*) Py_None()) could_not_convert!(T)(o);
         return;
+    } else static if (isSlice!T) {
+        import mir.ndslice.connect.cpython : toPythonBuffer, fromPythonBuffer, PythonBufferErrorCode, PyBuf_indirect, PyBuf_format, PyBuf_writable, bufferinfo;
+        enum PyBuf_full = PyBuf_indirect | PyBuf_format | PyBuf_writable;
+        if(PyObject_CheckReadBuffer(o) != -1) {
+            T x;
+            bufferinfo buf;
+            PyObject_GetBuffer(o, cast(Py_buffer*) &buf, PyBuf_full);
+            scope(exit) PyBuffer_Release(cast(Py_buffer*) &buf);
+            const auto err = fromPythonBuffer(x, buf);
+            if (err == PythonBufferErrorCode.success)
+            {
+                return x;
+            }
```

Comment:
Need to throw if unsuccessful

Reply:
It will do that automatically, after trying other conversions if possible